content/blog/2018-09-14-smapr.md

```diff
@@ -0,0 +1,637 @@
+weekly_smap <- projectRaster(weekly_smap, crs = projection(study_area))
+
+weekly_precip <- stack(weekly_precip)
+names(weekly_precip) <- start_dates
+weekly_precip <- resample(weekly_precip, weekly_smap)
+```
+
+Now that we have RasterStack objects, we will create tidy data frames that we can use in ggplot2.
+Because the code to do this is the same for both objects, I'll write a little helper function.
+
+
+```r
+make_study_area_df <- function(raster) {
+  raster %>%
+      mask(study_area) %>%
+      trim %>%
+      as('SpatialPixelsDataFrame') %>%
+      as.data.frame() %>%
+      gather(date, value, -x, -y) %>%
+      as_tibble %>%
+      mutate(date = start_dates[as.numeric(as.factor(date))])
+}
+
+soil_moisture <- make_study_area_df(weekly_smap) %>%
+  rename(sm = value)
+precip <- make_study_area_df(weekly_precip) %>%
+  rename(pr = value)
+```
+
+Let's check these out!
+
+
+```r
+soil_moisture
+```
+
+```
+## # A tibble: 171,124 x 4
+##        x     y date          sm
+##    <dbl> <dbl> <date>     <dbl>
+##  1  50.7  11.9 2015-04-01 0.185
+##  2  51.1  11.9 2015-04-01 0.127
+##  3  50.3  11.6 2015-04-01 0.337
+##  4  50.7  11.6 2015-04-01 0.150
+##  5  51.1  11.6 2015-04-01 0.139
+##  6  43.2  11.3 2015-04-01 0.340
+##  7  48.5  11.3 2015-04-01 0.557
+##  8  48.8  11.3 2015-04-01 0.361
+##  9  49.2  11.3 2015-04-01 0.398
+## 10  49.6  11.3 2015-04-01 0.150
+## # ... with 171,114 more rows
+```
+
+So we have a data frame where each row is a pixel with a date corresponding to the first day of the week, and then the mean soil moisture for that week in the `sm` column.
+What do these data look like?
+
+
+```r
+my_theme <- theme_minimal() +
+  theme(panel.grid.minor = element_blank())
+
+soil_moisture %>%
+  ggplot(aes(date, sm)) +
+  geom_point(alpha = .02) +
+  my_theme +
+  xlab('') +
+  ylab('Mean soil moisture (m^3 water per m^3 soil)')
+```
+
+![](/img/blog-images/2018-09-14-smapr/sm-timeseries-1.png)<!-- -->
+
+What about the precipitation data?
+
+
+```r
+precip %>%
+  ggplot(aes(date, pr)) +
+  geom_point(alpha = .02) +
+  my_theme +
+  xlab('') +
+  ylab('Mean precipitation (mm)')
+```
+
+![](/img/blog-images/2018-09-14-smapr/pr-timeseries-1.png)<!-- -->
+
+What does the relationship between precipitation and soil moisture look like?
+
+
+```r
+soil_moisture %>%
+  left_join(precip) %>%
+  ggplot(aes(pr, sm)) +
+  geom_point(alpha = .02) +
+  my_theme +
+  ylab('Mean soil moisture (m^3 water per m^3 soil)') +
+  xlab('Mean precipitation (mm)') +
+  scale_y_log10() +
+  scale_x_log10()
+```
+
+```
+## Joining, by = c("x", "y", "date")
+```
+
+![](/img/blog-images/2018-09-14-smapr/pr-vs-sm-1.png)<!-- -->
+
+Notice the band of zero-precipitation points that sit on the y-axis.
+Among nonzero precipitation values, it seems like there is a nonlinear relationship between precipitation and soil moisture.
+
+So, now let's take a look at the spring and summer of 2018, when the flooding was worst in Somalia and Kenya.
```

Comment:
minor comment: with the paired plots that follow where you have first soil moisture, then precipitation, it took me a bit to see that the plots were paired and the fig legend told me what the plot was. If it doesn't change the flow much, perhaps you could add "Soil moisture" or "Precipitation" in regular text above each.

Reply:
Done!